adapter/methods.ts

```diff
@@ -43,4 +53,91 @@ export async function addKusamaAsset(): Promise<void> {
             method: 'addKusamaAsset'
         }]
     });
-}
\ No newline at end of file
+}
+
+export async function addPolkadotAsset(): Promise<void> {
+    return await window.ethereum.send({
+        method: pluginOrigin,
+        params: [{
+            method: 'addPolkadotAsset'
+        }]
+    });
+}
+
+async function sendSnapMethod(method: 'getBalance' | 'getAddress' | 'getPublicKey' | 'exportSeed'): Promise<string|null> {
+    try {
+        return await window.ethereum.send({
+            method: pluginOrigin,
+            params: [{
+                method: method
+            }]
+        });
+    } catch (e) {
+        console.log("Keys not generated", e);
+    }
+    return null;
+}
+
+export async function getBalance(): Promise<string> {
+    const response = await sendSnapMethod("getBalance");
+    return (response != null) ? response : "Unable to fetch balance";
+}
+
+export async function getAddress(): Promise<string> {
+    const response = await sendSnapMethod("getAddress");
+    return (response != null) ? response : "Unable to fetch address";
+}
+
+export async function getPublicKey(): Promise<string> {
+    const response = await sendSnapMethod("getPublicKey");
+    return (response != null) ? response : "Unable to fetch public key";
+}
+
+export async function exportSeed(): Promise<string> {
+    const response = await sendSnapMethod("exportSeed");
+    return (response != null) ? response : "Unable to fetch seed";
+}
+
+
+export async function getLatestBlock(): Promise<BlockInfo> {
+    try {
+        return await window.ethereum.send({
+            method: pluginOrigin,
+            params: [{
+                method: "getBlock",
+                params: {
+                    blockTag: "latest"
+                }
+            }]
+        });
+    } catch (e) {
+        console.log("Unable to fetch latest block", e);
+        return {number: "", hash: ""};
+    }
+}
+
+export async function setConfiguration(configuration: SnapConfig): Promise<void> {
+    return await window.ethereum.send({
+        method: pluginOrigin,
+        params: [{
+            method: 'configure',
+            params: {configuration: configuration}
+        }]
+    });
+}
+
+export async function getAllTransactions(address?: string): Promise<any> {
```

Comment:
you should add return type to all responses, but I suggest you rebase of monorepo branch so you can add all interfaces and types in single place and share with other components